src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -1098,18 +1098,23 @@ private static void KillChildProcesses(int parentId, ProcessWithParentId[] curre
         [ArchitectureSensitive]
         private static bool IsWindowsApplication(string fileName)
         {
-#if UNIX
-            return false;
-#else
-            if (!Platform.IsWindowsDesktop) { return false; }
+            var fileInfo = new FileInfo(fileName);
+            if (!fileInfo.Exists)
+            {
+                throw new ArgumentException(
+                    StringUtil.Format(SessionStateStrings.PathNotFound, fileName));
+            }
```

Comment:
We could use  `ResolveLinkTarget()` - it throws if `fileName` doesn't exist.

Reply:
We cannot use `ResolveLinkTarget` because it calls `FileSystem.GetImmediateLinkTarget` with `returnFullPath: true`, which construct the full path [using the process working directory](https://github.com/dotnet/runtime/blob/98c05df043629d6aab85f34907742eab88e5d6da/src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Windows.cs#L517-L520) in some cases. Our existing code doesn't do that, and even if it should do that, it should use the current working directory in PowerShell instead of the process' working directory. So, the behavior difference cannot be ignored.

On the other hand, `GetLinkTarget` calls `FileSystem.GetImmediateLinkTarget` with `returnFullPath: false`. It has the exact same behavior as our existing code, except for throwing on non-existing file, which can be covered by a simple check easily.

Reply:
> which construct the full path using the process working directory in some cases

I don't see this in the implementation.

Reply:
You are right. I must be confused the code here with code from another place. Here it used the parent folder of the link path, which is correct when resolving the path.

However, when reviewing this `IsWindowsApplication` code again, I think we have a problem now. The `GetTarget` call was added in #13481 to handle Windows Store apps. However, we don't resolve target for `AppExecLink` anymore due to undocumented API (#16044), so the return value of `WinInternalGetTarget(fileName)` will always be `null` for Windows Store Apps, and thus the code added here are useless now ...

What should we do with it? /cc @SteveL-MSFT

Reply:
Had an offline discussion with @SteveL-MSFT, and the regression will have to be by-design for 7.2, because to retrieve the target for `AppExecLink`, we have to use undocumented APIs, which is not allowed.

We will start a conversation with the Windows App team and see if they can have the API documented so that we can have a better experience. But that will be post-7.2.

Reply:
I updated the change in `IsWindowsApplication` to use `ResolveLinkTarget(returnFinalTarget: false)`.
This is a more of a safe, back-compat change. Ideally, we should resolve a symbolic link earlier, and use the final target to start the process, both in `NativeCommandProcessor` and `Start-Process`. As of today, pwsh doesn't use the final target of a symbolic link when starting process, see #16171.